features/steps/steps.py

```diff
@@ -225,6 +288,18 @@ def step_impl(context, field, values):
 
     context.applicable = getattr(context, 'applicable', True) and applicable
 
+@given('{attr} forms {closed_or_open} curve')
+def step_impl(context, attr, closed_or_open):
+    assert closed_or_open in ('a closed', 'an open')
+    should_be_closed = closed_or_open == 'a closed'
+    try:
```

Comment:
Played with it for quite a while, doesn't quite work, although I loved the idea:

```
behave.step_registry.AmbiguousStep: @given('It forms {closed_or_open} curve') has already been defined in
existing step @given('{attr} forms {closed_or_open} curve') at features/steps/steps.py:289
```
I suppose we could perhaps use `behave.use_step_matcher("re")` and define `@given` using regex, which would allow the exclusion of `It ` in one of the steps. Something like below:

```
use_step_matcher("re")
@given(r"(It) forms (.+) curve") # looks for a string like: "It forms {an open} curve"
@given(r"(\b(?!It\b)\w+) forms (.+) curve") # looks for a string like: "Anything but 'It' forms {an open} curve"

```
I've tested it and it works, but in my opinion, it's not too readable. That's why I left the `@given` as was but exchanged the `try `for `if attr == 'It'`. What do you think?

Reply:
Also reformated due to a different comment, I would say it's more readable now. 

Multiple decorators is an useful tool, could come in handy.